data-prepper-plugins/elasticsearch/src/main/java/com/amazon/dataprepper/plugins/sink/elasticsearch/ElasticsearchSink.java

```diff
@@ -158,16 +170,65 @@ private void createIndexTemplate(final boolean isISMEnabled, final String ismPol
     if (isISMEnabled) {
       IndexStateManagement.attachPolicy(esSinkConfig.getIndexConfiguration(), ismPolicyId, indexAlias);
     }
+
     putIndexTemplateRequest.source(esSinkConfig.getIndexConfiguration().getIndexTemplate());
     restHighLevelClient.indices().putTemplate(putIndexTemplateRequest, RequestOptions.DEFAULT);
   }
 
+  // TODO: Unit tests for this (and for the rest of the class)
+  private boolean shouldCreateIndexTemplate(final String indexTemplateName) throws IOException {
+    final Optional<IndexTemplateMetadata> indexTemplateMetadataOptional = getIndexTemplateMetadata(indexTemplateName);
+    if (indexTemplateMetadataOptional.isPresent()) {
+      final Integer existingTemplateVersion = indexTemplateMetadataOptional.get().version();
+      LOG.info("Found version {} for existing index template {}", existingTemplateVersion, indexTemplateName);
+
+      final int newTemplateVersion = (int) esSinkConfig.getIndexConfiguration().getIndexTemplate().getOrDefault("version", 0);
+
+      if (existingTemplateVersion != null && existingTemplateVersion >= newTemplateVersion) {
+        LOG.info("Index template {} should not be updated, current version {} >= existing version {}",
+                indexTemplateName,
+                existingTemplateVersion,
+                newTemplateVersion);
+        return false;
+
+      } else {
+        LOG.info("Index template {} should be updated from version {} to version {}",
+                indexTemplateName,
+                existingTemplateVersion,
+                newTemplateVersion);
+        return true;
+      }
+    } else {
+      LOG.info("Index template {} does not exist and should be created", indexTemplateName);
+      return true;
+    }
+  }
+
+  private Optional<IndexTemplateMetadata> getIndexTemplateMetadata(final String indexTemplateName) throws IOException {
+    final IndexTemplatesExistRequest existsRequest = new IndexTemplatesExistRequest(indexTemplateName);
+    final boolean exists = restHighLevelClient.indices().existsTemplate(existsRequest, RequestOptions.DEFAULT);
```

Comment:
Instead of making a separate request to check existence, can we check response.getIndexTemplates().size() == 0?

Reply:
IIRC I got an ES exception because of the 404 returned when the template does not exist. Figured the "correct" way would be to make a separate "exists" call here. Not especially concerned with the extra call as this only happens on startup.

Reply:
Sure. Not blocking